bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java

```diff
@@ -191,6 +209,52 @@ public String toString() {
         }
     }
 
+    class ParallelReadRequest extends LedgerEntryRequest {
+
+        int numPendings;
+
+        ParallelReadRequest(ArrayList<InetSocketAddress> ensemble, long lId, long eId) {
+            super(ensemble, lId, eId);
+            numPendings = writeSet.size();
+        }
+
+        @Override
+        void read() {
+            for (int bookieIndex : writeSet) {
+                InetSocketAddress to = ensemble.get(bookieIndex);
+                try {
+                    sendReadTo(to, this);
+                } catch (InterruptedException ie) {
+                    LOG.error("Interrupted reading entry {} : ", this, ie);
+                    Thread.currentThread().interrupt();
+                    fail(BKException.Code.InterruptedException);
```

Comment:
yes. we don't fail the request when encountering a read failure (see logErrorAndReattemptRead). we fail here is because when you received a "InterrupteException", it means the thread is interrupted. it is most likely happening during closing the client, so we fail the request immediately.